### PR TITLE
Use interpolated strings in Control.cs

### PR DESCRIPTION
**Pull request by halgab**
Contributes to #8203

As the issue mentions that the PR sizes should be manageable, I chose to start with one file that sums up all the cases I encountered. I have plenty more changes ready to submit.

## Test methodology

static code analysis

## Test environment(s)

8.0.100-preview.1.23115.2, but I failed to run the tests locally...

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8681)

### CODE REVIEW

**Comment by halgab** [comment]
@dotnet-policy-service agree

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 1]
```diff
@@ -9758,17 +9758,8 @@ internal virtual void RecreateHandleCore()
...
-                {
-                    Debug.Write("RecreateHandle: ");
-                    Debug.Write(GetType().FullName);
-                    Debug.Write(" [Text=");
-                    Debug.Write(Text);
-                    Debug.Write("]");
-                    Debug.WriteLine("");
-                }
-#endif
+                Debug.WriteLineIf(CoreSwitches.PerfTrack.Enabled, $"RecreateHandle: {GetType().FullName} [Text={Text}]");
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:9761]: nice!

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 2]
```diff
@@ -11382,13 +11360,9 @@ public void SuspendLayout()
...
 
-#if DEBUG
             Debug.Assert(LayoutSuspendCount > 0, "SuspendLayout: layoutSuspendCount overflowed.");
-            if (CompModSwitches.LayoutSuspendResume.TraceInfo)
-            {
-                Debug.WriteLine($"{GetType().Name} ::SuspendLayout( newCount = {LayoutSuspendCount}");
-            }
-#endif
+            Debug.WriteLineIf(CompModSwitches.LayoutSuspendResume.TraceInfo,
+                $"{GetType().Name} ::SuspendLayout( newCount = {LayoutSuspendCount}");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11365]:
```suggestion
                $"{GetType().Name} ::SuspendLayout(newCount = {LayoutSuspendCount}");
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 3]
```diff
@@ -11508,11 +11482,9 @@ protected void UpdateBounds(int x, int y, int width, int height, int clientWidth
 #if DEBUG
             if (CompModSwitches.SetBounds.TraceVerbose)
             {
-                Debug.WriteLine(string.Format(CultureInfo.CurrentCulture, "{0}::UpdateBounds(", Name));
+                Debug.WriteLine($"{Name}::UpdateBounds(");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11485]:
```suggestion
                Debug.WriteLine($"{Name ?? ToString()}::UpdateBounds(");
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 4]
```diff
@@ -10328,13 +10315,8 @@ internal void ScaleControl(SizeF includedFactor, SizeF excludedFactor, Control r
                 {
                     excludedSpecified |= (~RequiredScaling & BoundsSpecified.All);
                 }
-#if DEBUG
-                if (CompModSwitches.RichLayout.TraceInfo)
-                {
-                    Debug.WriteLine(string.Format(CultureInfo.InvariantCulture, "Scaling {0} Included: {1}, Excluded: {2}",
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:10334]: I don't see need for Invariant Culture here. We should be okay with interpolation.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 5]
```diff
@@ -11536,7 +11508,11 @@ protected void UpdateBounds(int x, int y, int width, int height, int clientWidth
 #if DEBUG
                 if (Bounds != originalBounds && CompModSwitches.SetBounds.TraceWarning)
                 {
-                    Debug.WriteLine(string.Format(CultureInfo.CurrentCulture, "WARNING: Bounds changed during OnLocationChanged()\r\nbefore={0} after={1}", originalBounds, Bounds));
+                    Debug.WriteLine($"""
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11511]: nit: can be converted into `Debug.WriteLinIf`. Same below.

Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11511]: I thought so at first, but `originalBounds` only exists in debug...

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 4 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:10334]: I hadn't notice this one was invariant... I'll check again

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 2 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11365]: While we're trying to fix this (although I had doubts including this change as it's slightly out of scope...), doesn't it look like it's also missing a closing parenthesis?

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 3 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11485]: `Name` appears to be non-nullable though

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 4 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:10334]: I think it does have an impact on the output. The original issue mentioned the fact that it's not that relevant for debug statements anyway, but I can replace the code with a version that accounts for the culture if you prefer.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 2 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11365]: Good point, add the closing parenthesis as well!

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 3 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11485]: Do you agree with me that the fallback on `ToString()` is not necessary as `Name` cannot be null?

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 6]
```diff
@@ -11536,7 +11508,11 @@ protected void UpdateBounds(int x, int y, int width, int height, int clientWidth
...
+                        WARNING: Bounds changed during OnLocationChanged()
+
+                        before={originalBounds} after={Bounds}
```
Comment (dmitrii-drobotov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11514]: There was only one line break previously, the empty line is redundant:
```suggestion
                        WARNING: Bounds changed during OnLocationChanged()
                        before={originalBounds} after={Bounds}
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 7]
```diff
@@ -10743,12 +10725,8 @@ public void SetBounds(int x, int y, int width, int height, BoundsSpecified speci
...
         protected virtual void SetBoundsCore(int x, int y, int width, int height, BoundsSpecified specified)
         {
-#if DEBUG
-            if (CompModSwitches.SetBounds.TraceInfo)
-            {
-                Debug.WriteLine(string.Format(CultureInfo.CurrentCulture, "{0}::SetBoundsCore(x={1} y={2} width={3} height={4} specified={5}", Name, x, y, width, height, specified));
-            }
-#endif
+            Debug.WriteLineIf(CompModSwitches.SetBounds.TraceInfo,
+                $"{Name}::SetBoundsCore(x={x} y={y} width={width} height={height} specified={specified}");
```
Comment (dmitrii-drobotov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:10729]: Could you add a missing closing parenthesis here?
```suggestion
                $"{Name}::SetBoundsCore(x={x} y={y} width={width} height={height} specified={specified})");
```

**Review by dmitrii-drobotov — COMMENTED** [review]
Just a couple nits, otherwise looks good!

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 6 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: You're right, thanks. Fixed line 11537 as well

**Review by dmitrii-drobotov — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 4 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:10334]: @halgab, Do you have sample of what difference you noticed with `InvariantCulture` for this statement?

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 3 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11485]: i see `getter` for name checks for null. We should be good.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11485]: >Do you agree with me that the fallback on ToString() is not necessary as Name cannot be null?

@halgab - Name can be an empty string, in that case it's better to print out type name via the ToString method.

Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:11485]: I can add a check for empty string in a follow-up PR if you want me to.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 4 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:10334]: I don't, it's just that `includedFactor` and `excludedFactor` are both of type `SizeF`. On first read, I thought this type was culture-dependant, but reading the code again it's not, sorry for the confusion

**Pull request merged**